stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/ChannelRepository.kt

```diff
@@ -109,6 +110,13 @@ internal class ChannelRepositoryImpl(
         }
     }
 
+    override suspend fun evictChannelMessage(cid: String, messageId: String) {
+        val channel = channelCache.get(cid)
+        if (channel != null) {
+            channel.messages = channel.messages.filter { it.id != messageId }
+        }
+    }
+
```

Comment:
If we don't do this, the following will happen:
1. Message list screen
2. A message is hard deleted
3. The message disappears
4. Channel list screen
5. Message list screen
6. Deleted message is displayed (Messages from message repository and channel repository are concatenated)

Reply:
What about just remove such channel from the cache? It make logic simpler and easier. Next time when someone will fetch channel by id the repo takes it from DB

Reply:
Thanks, great suggestion!